scripts/refactor-go-logging.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+
+# Sed script to refactor log.go import and logging statements in .go files to match new sytnax.
+# Note: the log.Error(x) -> log.FormatErrors([]error{x})
+# mapping is slightly fragile.
+# If "x" has nested parens or anything complicated, it may
+# not work.
+# Seems to work on the cases I have run across, but sometimes
+# this is only by accident.
+# I figure special cases can be fixed by hand.
+
+sed -E -e '
+:top
+    # Skip any substitutions on comment lines
+    #
+    /^[[:blank:]]*\/\//{
+        n
+        b top
+    }
+
+
+    # FATAL events are mapped like this:
+    #
+    # log.Event(ctx, string, log.FATAL, log.Error(something1), something2) -> log.Fatal(ctx, string, something1, something2)
+    #
+    /log\.Event.*log\.FATAL/{
+
+        # change the function name
+        s/log\.Event/log.Fatal/
+
+        # remove the severity
+        s/,[[:blank:]]*log.FATAL//
+
+        # extract the log.Error argument
+        s/log.Error\(([^)]*)\)/\1/
+
+        # and go to the next line
+        n
+        b top
+    }
+
+    # And ERROR events are similar:
+    #
+    # log.Event(ctx, string, log.ERROR log.Error(something1), something2) -> log.Error(ctx, string, something1, something2)
+    #
+    /log\.Event.*log\.ERROR/{
+
+        # remove the severity
+        s/,[[:blank:]]*log.ERROR//
+
+        # extract the log.Error argument
+        s/log.Error\(([^)]*)\)/\1/
+
+        # change the function name
+        s/log\.Event/log.Error/
+
+        # and go to the next line
+        n
+        b top
+    }
+
+    # Info events are mapped like this:
+    #
+    # log.Event(ctx, string, log.INFO, log.Error(something)) -> log.Info(ctx, string, log.FormatErrors([]string{something}))
```

Comment:
So INFO log event shouldn't contain an `log.Error` but will usually have `log.Data{}` as the last variable (arguably this is always optional for all logs). I guess the `log.Error` mapping is allowed by the log library even if we do not use it in our apps/libraries.

I assume the mapping for log.Data does not change as it is the last variable and is the same for both v1 and v2 logs?

Reply:
Yeah - @wavemechanics to confirm, but I don't think this will touch`log.Data{}` at all. Given the above, is the replacement of `log.Error` with `log.FormatError` needed on line 72?

Reply:
following offline discussion I think this is fine to leave as is.